atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -348,3 +365,126 @@ def get_signing_intent(self):
 
         signing_intent = odcs_config.get_signing_intent_by_name(self.signing_intent)
         return signing_intent
+
+    def _get_exclude_list(self, request_session, exclude_list_url):
+        response = request_session.get(exclude_list_url)
+        response.raise_for_status()
+        exclude_list_yaml = yaml.safe_load(response.text)
+        # prepend os.sep so it starts with exact match, glob.glob doesn't need it
+        return [os.path.join(os.sep, k, item) for k, v in exclude_list_yaml.items() for item in v]
+
+    def _create_full_remote_sources_map(self, request_session, remote_sources_map,
+                                        remote_sources_dir):
+        full_remote_sources_map = {}
+
+        for remote_source, remote_source_json in remote_sources_map.items():
+            remote_source_archive = os.path.join(remote_sources_dir, remote_source)
+
+            response = request_session.get(remote_source_json)
+            response.raise_for_status()
+            response_json = response.json()
+            full_remote_sources_map[remote_source_archive] = response_json
+        return full_remote_sources_map
+
+    def _check_if_package_excluded(self, remote_json, exclude_list, remote_archive):
+        delete_app = False
+        # check if any package in cachito json matches excluded entry
+        for package in remote_json['packages']:
+            if delete_app:
+                break
+
+            for exclude_path in exclude_list:
+                if package.get('name').endswith(exclude_path):
+                    self.log.debug('Package excluded: "%s" from "%s"', package.get('name'),
+                                   remote_archive)
+                    delete_app = True
+                    break
+        return delete_app
+
+    def _delete_app_directory(self, remote_source_dir, unpack_dir, remote_archive):
+        vendor_dir = os.path.join(remote_source_dir, unpack_dir, 'app', 'vendor')
+
+        if os.path.exists(vendor_dir):
+            shutil.move(vendor_dir, remote_source_dir)
+            self.log.debug('Removing app from "%s"', remote_archive)
+            shutil.rmtree(os.path.join(unpack_dir, 'app'))
```

Comment:
Shouldn't the path be `os.path.join(remote_source_dir, unpack_dir, 'app')`? Or am I missing something?

Reply:
remote_source_dir is the dir where those downloaded archives are and unpack_dir is 
archive name with pull path with _unpacked (just for unpacking)
I am moving vendor to main dir, so I can delete whole app dir in unpack_dir 
and then moving it back

Reply:
Is `unpack_dir` a subdirectory of `remote_source_dir`? I'm confused because
- `vendor_dir` uses `os.path.join(remote_source_dir, unpack_dir, 'app', 'vendor')`
- here it's just `os.path.join(unpack_dir, 'app')`

If it is a subdirectory, it would be nice to keep it consistent

Reply:
remote_source_dir & unpack_dir   are full path dirs 

this is creating confusion :          vendor_dir = os.path.join(remote_source_dir, unpack_dir, 'app', 'vendor')

it works BUT it should be just:

vendor_dir = os.path.join(unpack_dir, 'app', 'vendor')

os.path.join just joins those dirs with full path in mind so it does it works

I'll change this